Lib/asyncio/sslproto.py

```diff
@@ -461,7 +461,7 @@ def eof_received(self):
                 logger.debug("%r received EOF", self)
 
             if self._state == SSLProtocolState.DO_HANDSHAKE:
-                self._on_handshake_complete(ConnectionResetError)
+                self._on_handshake_complete(ConnectionResetError())
```

Comment:
Hi @mjpieters; This fix caused the ssl-over-ssl test from [GH-113214](https://github.com/python/cpython/issues/113214) (PR #113334) to fail and want to make sure I did not break anything on your end.

The test will raise `ConnectionResetError` (via `SSLProtocol.eof_received`) and normally this won't
be caught by the exception handler in [`SSLProtocol._fatal_error` because it is of instance `OSError`](https://github.com/python/cpython/blob/dce30c9cbc212e5455e100f35ac6afeb30dfd23e/Lib/asyncio/sslproto.py#L916-L925). But because we no longer raise the exception in `SSLProtocol._on_handshake_complete(exc)`, the exception object is never initialised and the `isinstance` doesn't work as intended. Changed `SSLProtocol.eof_received` to pass an exception object and not class to fix this.

Reply:
I'm not an expert in this area, but passing in an instance instead of the class looks fine by me. I did leave a comment on the removal of the `try...except` block however as this did more than just handle `raise handshake_exc`.